src/image/image_type.hpp

```diff
@@ -9,6 +9,7 @@
 #include <string>
 #include <serialization/serialization_includes.hpp>
 #include <boost/shared_ptr.hpp>
+#include <boost/gil/gil_all.hpp>
```

Comment:
Make sure you add boost to the CMakeLists.txt requires list

Reply:
Since we were already using <boost/shared_ptr.hpp>, wouldn't it already be there?

Reply:
Eh, check -- I don't think it's there now. 

With header only types like shared pointer, the consequence of not having it there is pretty minimal (essentially, running make in this directory after a clean configure will fail).  But if it's not header only (as I think the GIL is), there can be annoying linker errors down the road.

Reply:
Can we avoid including boost gil in a header? This will slow down compilation massively. image_type.hpp is included in quite a lot of places.

Reply:
Given the code we can just forward declare boost::gil::rgb8_image_t

Reply:
I also call `const_view` on the `rgb8_image_t` as well as `boost::gil::num_channels`. I'm assuming I forward declare those too?

Reply:
Boost is automatically added as a dependency to all other (non-external) targets. The `make_library` and `make_executable` macros do this.

Reply:
@hoytak @ylow @znation Got rid of the `boost/gil/gil_all` import by making the constructor a template function that can take multiple Boost GIL image types. Please let me know if that looks good!

Reply:
Hmm, looks like it fails the CI: https://travis-ci.com/apple/turicreate/jobs/197093292

Not really sure why...

Reply:
Pushed a fix for the build failure.